lib/safira/accounts/accounts.ex

```diff
@@ -102,12 +110,25 @@ defmodule Safira.Accounts do
     end
   end
 
-  def get_attendee(id) do
-    Repo.get(Attendee, id)
+  def get_attendee_with_badge_count_by_username!(username) do
+    case get_attendee_by_username!(username) do
+      nil ->
+        nil
+
+      %Attendee{} = attendee ->
+        badges =
+          attendee.badges
+          |> Enum.filter(&(&1.type != 0))
+
+        Map.put(attendee, :badge_count, length(badges))
+    end
+  end
```

Comment:
I think you should remove the `!` from the end of the function name, since it should only be used in functions that [throw an exception when they encounter an error](https://hexdocs.pm/elixir/1.12/naming-conventions.html#trailing-bang-foo) (in this case it just returns null). And change `nil` case.

```suggestion
  def get_attendee_with_badge_count_by_username(username) do
    case get_attendee_by_username!(username) do
      %Attendee{} = attendee ->
        badges =
          attendee.badges
          |> Enum.filter(&(&1.type != 0))
          
      _ ->
        nil
        Map.put(attendee, :badge_count, length(badges))
    end
  end
```